### ISSUE TITLE
Rule to handle errors

### ISSUE THREAD

**Issue by egeerardyn** [islandoftex/arara#78]
First of all, let me thank you for the great work on this tool! I use it with joy everyday. However, there are a few small annoyances that I have with it and this is the major one, for which I haven't found a decent work-around.

I'm currently writing my thesis, which is quite a complicated document such that if I make a syntax error, this introduces errors in my AUX file and hence every subsequent compilation fails even if my TeX files are all fine again.
Concretely, I would love to see a conditional pragma like `arara: clean {...} if error` that executes only if one of the previous rules fails.

My current work-around for this is to have the following in my main file:

``` latex
%% arara: clean: {...}
% arara: xelatex
% arara: ...
```

when an error happens, I temporarily remove one of the `%` signs, hit "Build" and replace it again to avoid that I have huge compilation times afterwards.

Practically, I think something like

``` latex
% arara: xelatex
% arara: ...
% arara: clean: {} if error
```

would be a great alternative for me.

If somebody knows an alternative (that works on arara 3, preferably), that would be great as well!

**Comment by samcarter8** [comment]
With 4.0 it is possible to search in files for error messages, so you could do something along the lines of 

```
% arara: clean: { files: [document.aux] }
% arara: --> if found('log', '==> Fatal error occurred')

% arara: pdflatex
```
(for me this feature was the reason to switch to 4.0)

**Comment by egeerardyn** [comment]
@samcarter8 Thank you for the hint! I will have a look at arara 4.0 (unfortunately, MacTeX still comes with 3.0).

**Comment by cereda** [comment]
Dear Egon,

Thank you very much for the kind words, I am very happy to hear `arara` is useful for you. :wink: 

First and foremost, I need to apologize for the huge delay in finally releasing the new version in the wild. Time is quite complicated for me at the moment, as I need to write my thesis. The current official release (3.0) has a lot of nuisances and critical bugs that were hopefully zapped in the upcoming version (4.0).

The new version has several improvements and new features, and mostly because of them, the new manual has to be rewritten entirely, from scratch. Sadly, for me, writing manuals take a huge amount of time and effort, as I am not a native speaker. Besides, I need to write a gentle introduction to all features, as TeX users have several spectra of knowledge and might not grasp the ideas at first.

As @samcarter8 suggested, this directive conditional might indeed help you. Note that the `if` conditional only executes the directive (in this case, `clean`) if, and only if, the logical expression holds true. So, if the `==> Fatal error occurred` text pattern is found in the log file, the `clean` directive will be triggered. Very clever!

The code is now (to some extent) frozen, but the rules are still subject to change. There is a lot of things I still want to include in them, but as they are now, I think they already offer some neat playground.  If you want, I can provide a binary of version 4.0 for you to play if (together with the new rules), so you use these new features. But I understand this might not be the best time of drastically changing workflows (your thesis, after all, comes first). :wink: 

Please, let me know if you need any further help or more detailed explanations. Also, @samcarter8, thank you for the help. :wink: 

All the best!

Paulo

**Comment by egeerardyn** [comment]
Hi Paulo,

All my respect for the thesis coming first :+1:. To quote Douglas Adams:time is an illusion, ~~lunchtime~~ thesis time doubly so.

If you do have a binary for OS X, that would be very welcome since I can't seem to get maven running on homebrew (and I don't want to sink too much time into that right now).

Thanks!

PS: Maybe after both of us finish writing, we should have a chat about `arara` and whether I can help out a bit with documentation/coding/...

**Comment by egeerardyn** [comment]
Gentle ping for @cereda: If you do have some binary available for arara 4.0, I am still very interested.
